src/fmt.rs

```diff
@@ -1,25 +1,42 @@
 use crate::chain::Chain;
 use crate::error::ErrorImpl;
-use core::fmt::{self, Debug, Write};
+use core::fmt::{self, Write};
 
 impl ErrorImpl<()> {
     pub(crate) fn display(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}", self.error())?;
-
-        if f.alternate() {
-            for cause in self.chain().skip(1) {
-                write!(f, ": {}", cause)?;
-            }
-        }
+        let error = self.error();
+        let handler = self.handler();
 
-        Ok(())
+        handler.display(error, f)
     }
 
     pub(crate) fn debug(&self, f: &mut fmt::Formatter) -> fmt::Result {
         let error = self.error();
+        let handler = self.handler();
+
+        handler.debug(error, f)
+    }
+}
+
+impl crate::ReportHandler for crate::DefaultHandler {
+    #[cfg(backtrace)]
+    fn backtrace<'a>(
```

Comment:
I'm not sure we can have a publicly exported method that depends on a non feature `cfg`, I'll make sure to test this when I port `color-eyre` to depend on this branch